Add Beta Testing page

claudiucelfilip opened the pull request:
#### Changes proposed in this Pull Request

More info: pd2n8V-46-p2

This adds the Beta Testing page which showcases info about the Horizon programme.
The menu link is added to the Public API via D63456-code and #54588 (in the fallback menu).

Proposed design:
<img width="480" alt="beta-testing-screen-v8" src="https://user-images.githubusercontent.com/2749938/126165669-11b2e27c-c2ab-4487-ab66-64b7e57d0481.jpg">

#### Testing instructions

* Go to the /beta-testing page
* The page should render properly on any screen size

<img width="480" alt="Screenshot on 2021-07-21 at 14-05-25" src="https://user-images.githubusercontent.com/2749938/126480958-98b70ef4-f967-4f8f-b95e-65c20759fc93.png">
<img width="320" alt="Screenshot on 2021-07-21 at 14-05-48" src="https://user-images.githubusercontent.com/2749938/126480954-672b45f9-28bc-4937-b85c-a97b08001f0d.png">
<img width="249" alt="Screenshot on 2021-07-21 at 14-08-28" src="https://user-images.githubusercontent.com/2749938/126480951-b269a30d-bbb8-442e-af86-e9d72f7f0933.png">

* The `Join the Beta Program` link should go to horizon.wordpress.com
* The `Learn how it works` link should go to the About page on horizonfeedback.wordpress.com
* The `Recent Beta Features` should populate with the last 3 posts from horizonfeedback.wordpress.com

* There should be no related JS errors

claudiucelfilip committed:
Adds simple beta testing page

claudiucelfilip committed:
Adds some temporary main content

claudiucelfilip committed:
Adds more content and recent features section

claudiucelfilip committed:
Add nav link to fallback menu

claudiucelfilip committed:
Makes small improvements to the content

claudiucelfilip committed:
Changes the nav link icon

claudiucelfilip committed:
Adds a small code formatting fix

a8ci18n commented on client/my-sites/sidebar-unified/static-data/fallback-menu.js:
```diff
@@ -573,6 +573,13 @@ export default function buildFallbackResponse( {
+		{
+			icon: 'dashicons-hammer',
+			slug: 'beta-testing',
+			title: translate( 'Beta Testing' ),
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **122** times:
`translate( 'Beta Testing', { context: 'Plugin Installer'} )` *ES Score: 16*

a8ci18n commented on client/my-sites/beta-testing/main.jsx:
```diff
@@ -0,0 +1,187 @@
+
+			<Card>
+				<CardHeading tagName="h1" size={ 21 }>
+					{ translate( 'How it Works' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **10** times:
`translate( 'How it works.' )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/beta-testing)

claudiucelfilip committed:
Applies new design and copy

claudiucelfilip committed:
Apply latest design tweaks

claudiucelfilip committed:
Replaces fallback icon

claudiucelfilip committed:
Removes unused config entry

claudiucelfilip committed:
Removes fallback menu link to page

a8ci18n commented on client/my-sites/beta-testing/main.jsx:
```diff
@@ -0,0 +1,194 @@
+	const header = (
+		<FormattedHeader
+			brandFont
+			headerText={ translate( 'Beta Testing' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **122** times:
`translate( 'Beta Testing', { context: 'Plugin Installer'} )` *ES Score: 16*

claudiucelfilip committed:
Reuse available translation string

a8ci18n commented on client/my-sites/beta-testing/main.jsx:
```diff
@@ -0,0 +1,194 @@
+	const header = (
+		<FormattedHeader
+			brandFont
+			headerText={ translate( 'Beta Testing', { context: 'Plugin Installer'} ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **122** times:
`translate( 'Beta Testing', { context: 'Plugin Installer'} )` *ES Score: 16*

claudiucelfilip committed:
Makes small layout improvements

a8ci18n commented on client/my-sites/beta-testing/main.jsx:
```diff
@@ -0,0 +1,194 @@
+	const header = (
+		<FormattedHeader
+			brandFont
+			headerText={ translate( 'Beta Testing', { context: 'Plugin Installer' } ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **122** times:
`translate( 'Beta Testing', { context: 'Plugin Installer'} )` *ES Score: 16*

claudiucelfilip replied:
Done as suggested, but had to add an extra space before the end-curly braces.

claudiucelfilip committed:
Adds small font fix for the see all posts link

claudiucelfilip committed:
Applies latest style updates

claudiucelfilip committed:
Remove unused dependencies

a8ci18n commented on client/my-sites/beta-testing/main.jsx:
```diff
@@ -0,0 +1,188 @@
+					<img
+						className="beta-testing__main-image"
+						src="/calypso/images/illustrations/illustration-404.svg"
+						alt={ translate( 'Beta Testing' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **122** times:
`translate( 'Beta Testing', { context: 'Plugin Installer'} )` *ES Score: 16*

claudiucelfilip committed:
Reuse available translation string

donlair commented:
@claudiucelfilip this works as expected with no errors and in all screen resolutions I tested.

Couple things jumped out at me:
* Should we implement this using `react-query` instead of Redux? p4TIVU-9GP-p2 for context and example implementation at #53207.
* Lodash is deprecated in Calypso (p4TIVU-9Bf-p2)
* It would be nice if we could keep people in Calypso rather than send them to the Horizon blog. Maybe we could use something like the [InlineSupportLink](https://github.com/Automattic/wp-calypso/tree/trunk/client/components/inline-support-link) component to make the content available in a modal.
* Should I receive some confirmation that I've joined the program when I click "Join the Beta Program"?

claudiucelfilip committed:
Removes lodash dependency

claudiucelfilip committed:
Use react-query to fetch horizon posts

danielwrobert reviewed (approved):
LGTM! 🚀 

Hopefully there will be more posts soon to populate the RECENT BETA FEATURES section - the last three showing are form 2017 & 2018. 😜

olaolusoga commented:
Looking good.

Question: when the left navigation is white, is it the default to not have a 1px border-right that shows where the navigation ends?

olaolusoga commented:
Also, notes on tablet and mobile view. Can we re-add the image into all viewports. Below are screenshots of what that would look like. Thanks

Portrait tablet
<img width="697" alt="Screen Shot 2021-07-20 at 10 39 11 PM" src="https://user-images.githubusercontent.com/1382943/126422306-d57fa2f1-dfd1-4b87-bbf7-aaa8aacee246.png">

Mobile
<img width="388" alt="Screen Shot 2021-07-20 at 10 41 19 PM" src="https://user-images.githubusercontent.com/1382943/126422336-aeeb499a-a1a7-4831-959b-0c06cdbd6a56.png">

claudiucelfilip committed:
Maintains the main image on table and mobile views

claudiucelfilip committed:
Avoids using deprecated breakpoint includes

claudiucelfilip commented:
@donlair Thanks for the review and suggestions.
> * Should we implement this using `react-query` instead of Redux? p4TIVU-9GP-p2 for context and example implementation at [My Home: simplify My Home data layer by migrating to React Query #53207](https://github.com/Automattic/wp-calypso/pull/53207).

This looks like a nice approach! I've integrated it to fetch the posts, based on the examples.

> * Lodash is deprecated in Calypso (p4TIVU-9Bf-p2)

Removed it. It was used for a function we didn't really need.

> * It would be nice if we could keep people in Calypso rather than send them to the Horizon blog. Maybe we could use something like the [InlineSupportLink](https://github.com/Automattic/wp-calypso/tree/trunk/client/components/inline-support-link) component to make the content available in a modal.

I played around with InlineSupportLink but doesn't really work with non-support link. 
Outside of that, this seems like a 'nice to have' which we can add in another revision?

> * Should I receive some confirmation that I've joined the program when I click "Join the Beta Program"?

AFAIK there's no flag or member list you adhere to once you click that link. It just takes you to `horizon.wordpress.com` and you're in the program as long as you use that domain (instead of `wordpress.com`) for managing your site(s). 
On the other hand, I don't see why we couldn't send an email for first-time clicks on the button, but maybe in an upcoming revision?

claudiucelfilip commented:
@olaolusoga 
> Also, notes on tablet and mobile view. Can we re-add the image into all viewports. Below are screenshots of what that would look like. Thanks

Thanks for these mockups. Keeping the images seems like a better approach. I've added them back for these views and updated the screenshots in the descriptions.

> Question: when the left navigation is white, is it the default to not have a 1px border-right that shows where the navigation ends?

I also though about this, initially, and added a border between them. But it then visited the `/home` page and realised it also is full white but has no border, so I removed for the Beta Testing page.

<img width="480" alt="Screenshot on 2021-07-21 at 13-11-11 (1)" src="https://user-images.githubusercontent.com/2749938/126481545-4069ab95-fc5d-4801-a012-92584c53ade3.png">

danielwrobert commented on client/my-sites/beta-testing/main.jsx:
```diff
@@ -0,0 +1,183 @@
+import { useTranslate } from 'i18n-calypso';
+import FollowButtonContainer from 'calypso/blocks/follow-button';
+import Gridicon from 'calypso/components/gridicon';
+import { useQuery } from 'react-query';
```
@claudiucelfilip I noticed that [the other instances](https://github.com/Automattic/wp-calypso/search?q=useQuery) of `useQuery` in this project are abstracted out and live in the `data` directory. Do we want to follow that same approach here for consistency?

FWIW, this is my first look at `react-query` so I may be missing something - an initial observation, really! 😄

claudiucelfilip replied:
That's good point, thanks! 
I added a more general posts query to the `data` folder, which can be used in other locations.

danielwrobert reviewed (commented):
Left [a comment above](https://github.com/Automattic/wp-calypso/pull/54346#pullrequestreview-712236648) on the placement of the query functionality.

Other than that potential consideration, everything else LGTM and tests as described!

I also noticed some e2e tests failing - which may be entirely unrelated but worth confirming. 🤪

claudiucelfilip committed:
Abstract post query outside the component

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6269486

Thank you @claudiucelfilip for including a screenshot in the description! This is really helpful for our translators.

claudiucelfilip committed:
Remove unused dependency

danielwrobert commented on client/my-sites/beta-testing/style.scss:
```diff
@@ -0,0 +1,205 @@
+@import '@automattic/onboarding/styles/mixins';
+
+body.is-section-beta-testing {
```
I think this rule may be the issue re: @olaolusoga's note on [the white navigation here](https://github.com/Automattic/wp-calypso/pull/54346#issuecomment-883838504).

Comparing with the `/help` page you'll notice that it's set to `rgb(246, 247, 247)` there while this rule is explicitly setting it to `#fff`, which is then blending in with the white navigation on the  **Classic Bright** theme.

**Help Page:**
<img width="1421" alt="CleanShot 2021-07-22 at 11 41 56@2x" src="https://user-images.githubusercontent.com/481776/126669281-9fe0c5b5-c402-4517-8ec0-cd29f81c1ef4.png">

Beta Testing Page:
<img width="1422" alt="CleanShot 2021-07-22 at 11 40 56@2x" src="https://user-images.githubusercontent.com/481776/126667983-3fd2a39a-8d4a-4e52-8996-5b8049cb6cd0.png">

Would it be possible to remove this and, instead, apply the white background to the element with the `#primary` / `.layout__primary`?

**Example**
<img width="1420" alt="CleanShot 2021-07-22 at 11 44 06@2x" src="https://user-images.githubusercontent.com/481776/126668629-90893cbe-a8f8-44bd-bafe-49bfbbecb12c.png">

What do y'all think?

claudiucelfilip replied:
I don't have a strong opinion on this, but we did have a previous design iteration which looked similar.

<img width="480" alt="Screenshot on 2021-07-15 at 11-30-46" src="https://user-images.githubusercontent.com/2749938/126772178-23ac401e-9e78-4e7e-9a80-8fcfff136c07.png">

Happy to return to that version (or similar) if this is the desired approach.

danielwrobert replied:
Oh, I wasn't aware of this version. Personally, I think that previous iteration is better - it seems more consistent with the other pages in the admin and it solves the white background issue on the Classic Bright theme. But I'll defer to @olaolusoga on this for the final call!

I'd also be in favor of merging this and readdressing the design as a follow-up, if we're not certain/clear on what option we'd prefer.

danielwrobert replied:
@claudiucelfilip I spoke w/ @olaolusoga earlier today and he advised that white background is preferred. He also advised adding a border as briefly discussed above.

I recommend adding the border to the right side of the navigation when the navigation is white, based on the active theme (which I believe only occurs on the Classic Bright theme), as opposed to the content area. This way we don't have to worry about addressing multiple pages/components.

We can circle back and look at this in a follow-up. Let's go ahead and ship this as-is, when the translations complete.

danielwrobert reviewed (commented):
@claudiucelfilip I left some additional thoughts on the background color [here](https://github.com/Automattic/wp-calypso/pull/54346#discussion_r674935281).

Aside from that consideration, the updates LGTM! Thanks for abstracting the query - it seems more flexible now! 🎉

olaolusoga commented:
Making it all white is intentional. This page is an experiment, with the hope off all other pages following shortly.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6269486).

danielwrobert committed:
Remove space between imports.

This silences the eslint import/order rule that is causing a failing
TeamCity build.
